xwiki-pro-macros-api/src/main/java/com/xwiki/macros/excerptinclude/internal/ExcerptIncludeMacroRunnable.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.macros.excerptinclude.internal;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.XDOM;
+import org.xwiki.rendering.block.match.MacroBlockMatcher;
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Updates the excerpt-include macro reference parameter after the rename of an excerpt page.
+ *
+ * @version $Id$
+ * @since 1.14.4
+ */
+@Component(roles = ExcerptIncludeMacroRunnable.class)
+@Singleton
+@Unstable
+public class ExcerptIncludeMacroRunnable extends AbstractExcerptIncludeRunnable
+{
+    private static final String MACRO_REFERENCE_PARAMETER = "0";
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    @Named("compact")
+    private EntityReferenceSerializer<String> compactEntityReferenceSerializer;
+
+    @Inject
+    private Logger logger;
```

Comment:
This is hiding a field in the parent class and iirc in this case the component manager cannot inject the component. Did you test that this actually works?

Reply:
I tested it, and having another logger here has the advantage that it logs for the current class and not the base class.
Then I have to admit that I did not check of the logger in the base class is usable.